cmd/kubeadm/app/cmd/init.go

```diff
@@ -112,12 +113,30 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		"etcd client key file. Note: The path must be in /etc/ssl/certs",
 	)
 	cmd.PersistentFlags().MarkDeprecated("external-etcd-keyfile", "this flag will be removed when componentconfig exists")
+	cmd.PersistentFlags().BoolVar(
+		&skipChecks, "skip-checks", false,
+		"skip checks normally run before modifying the system",
+	)
 
 	return cmd
 }
 
 // RunInit executes master node provisioning, including certificates, needed static pod manifests, etc.
-func RunInit(out io.Writer, cmd *cobra.Command, args []string, cfg *kubeadmapi.MasterConfiguration) error {
+func RunInit(out io.Writer, cmd *cobra.Command, args []string, cfg *kubeadmapi.MasterConfiguration, skipChecks bool) error {
+	if u, ok := preflight.IsRoot(); !ok {
+		return &preflight.NotRootError{"<cmd/init> Requires root priviledges", u}
+	}
```

Comment:
Is it not possible for it to be in the normal list of checks, and still return it's own code?